### PR TITLE
Add TypeScript definition

### LINKED ISSUE #66: ES6 module syntax
It'd be great if we could get ES6 module syntax working, so it'd be possible to import this package using:
```
import { init as initDebugger } from 'electron-debug';
```

Which doesn't seem possible at the moment because the initialization function is exported via `module.exports = function() { ... }` and not as any named export.

The best solution would obviously be to `import initDebugger from 'electron-debug'` but that would break backwards compatibility which I assume we wouldn't want.

The change seems like something that could be made fairly quickly, so let me know if prefer me to submit a PR.

(For completeness, if you try `import foo from 'electron-debug'; foo({})` then Webpack will throw a warning.)

### PR DESCRIPTION
**Pull request by CvX**
Closes #66

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+ * });
+ */
+export default function electronDebug(options?: Options): void;
+
+/**
+ * Reload the specified `BrowserWindow` instance or the focused one.
+ *
+ * @param window - Default: BrowserWindow.getFocusedWindow()
+ */
+export function refresh(window?: Electron.BrowserWindow): void;
```
Comment (sindresorhus) [inline index.d.ts]: Couldn't you use a default parameter here?

```js
export function refresh(window?: Electron.BrowserWindow = BrowserWindow.getFocusedWindow()): void;
```

Then line 43 would be moot.

**Comment by sindresorhus** [comment]
Instead of using the `Electron` global, wouldn't it be better to explicitly import what we need?

```typescript
import {BrowserWindow} from 'electron';
```

File: index.d.ts (hunk 1 above)
Comment (CvX) [reply index.d.ts]: Hm, the default value doesn't seem to show up in the inline docs preview when using this syntax.

Comment (sindresorhus) [reply index.d.ts]: But it’s valid syntax, right? If so, I think we should still use it and open an issue on VSCode about it.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+ *
+ * electronDebug();
+ *
+ * let win;
+ * (async () => {
+ * 	await app.whenReady();
+ * 	win = new BrowserWindow();
+ * });
+ */
+export default function electronDebug(options?: Readonly<Options>): void;
```
Comment (sindresorhus) [inline index.d.ts]: Hmm, I think we should use `readonly` on the specific properties whenever possible, as `Readonly` will show up in the code completion and create noise: screen shot 2019-01-28 at 19 56 25

Maybe that's a non-issue. I dunno.

Thoughts?

---

*Really want [the `readonly` type operators](https://github.com/Microsoft/TypeScript/issues/10725)...

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+/**
+ * Reload the specified `BrowserWindow` instance or the focused one.
+ *
+ * @param [window]
```
Comment (sindresorhus) [inline index.d.ts]: No point in documenting these as they're already documented by the interface itself. I only do `params` when there's a description needed.

File: index.d.ts (hunk 2 above)
Comment (CvX) [reply index.d.ts]: I'm not sure either…
Is that information useful in any way to the end user of the API? If it's not, then sure, `readonly` all the props. 😉

Comment (sindresorhus) [reply index.d.ts]: It’s not. It’s only useful for us to ensure we don’t modify the object.

Comment (sindresorhus) [reply index.d.ts]: Let’s go with per-property annotation.

**Comment by sindresorhus** [comment]
https://github.com/sindresorhus/electron-debug/pull/68#discussion_r251277109 doesn't actually work...

I'm getting the error:

> TS2371: A parameter initializer is only allowed in a function or constructor implementation.

Seems it's not allowed in ambient context... Which sucks.

I ended up with this: https://github.com/sindresorhus/electron-debug/commit/142b2de52b946380d87ccd7b385d76691b90ed73#diff-b52768974e6bc0faccb7d4b75b162c99R52

I tried `@param [window=BrowserWindow.getFocusedWindow()]`, which is documented in the JSDoc docs, but that doesn't show anything in VSCode...

**Comment by sindresorhus** [comment]
I also realized another thing. There actually isn't really any point in using `readonly` for the input options after all. Seeing as the source where we use the options object is in JS, so the type definition isn't actually enforcing the read-only-ness... So it feels moot, unless I'm missing something?

**Comment by CvX** [comment]
It basically acts as a blueprint for the potential conversion to TypeScript. Or a source of "truth" about the code, even if not enforced by the tooling. 🤷‍♂️

**Comment by sindresorhus** [comment]
Hmm, yeah, I guess. Alright, let's keep it for now.

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * 	await app.whenReady();
+ * 	win = new BrowserWindow();
+ * });
+ */
+export default function electronDebug(options?: Options): void;
+
+/**
+ * Reload the specified `BrowserWindow` instance or the focused one.
+ */
+export function refresh(window?: BrowserWindow = BrowserWindow.getFocusedWindow()): void;
```
Comment (SamVerschueren) [inline index.d.ts]: Do you need to mark this as optional? Might be different in type definitions, but in TypeScript, assigning a default values makes it optional. You can't even do this in TS. But again, might be type definitions allowing that.

**Review by 0xsarawut — APPROVED** [review]

**Pull request merged**